### DISCUSSION TITLE
why does getSession return null on my Ubuntu server?

### DISCUSSION THREAD

**Discussion by Rongeegee** [vercel/next.js#45618] [Help] [answered]
I get the user id from session in `getServersidesprops` and use the id to get the data on the server side and return the props to my page component. I have the following in my file.

```
import { getSession } from 'next-auth/react';

.....
export async function getServerSideProps(context) {
    .................
    const session = await getSession(context);
    .................
}
```
On my remote server, session is null. My server is on Ubuntu 20. However, on my localhost and the server of Onrender, session has an expected value returned from getSession method. What is reason this is happening?

the following is part of the error message
```
  url:  '<based_url>/api/auth/session',
  message: 'Unexpected token < in JSON at position 0'
```

**Comment by simonecervini** [comment, marked as answer]
This is not a Next.js-related problem. I answered it here: https://github.com/nextauthjs/next-auth/discussions/6632?sort=old
